graph/iterator/and_iterator.go

```diff
@@ -31,13 +31,15 @@ type And struct {
 	result            graph.Value
 	runstats          graph.IteratorStats
 	err               error
+	qs                graph.QuadStore
 }
 
 // Creates a new And iterator.
-func NewAnd() *And {
+func NewAnd(qs graph.QuadStore) *And {
```

Comment:
It would be good to have a comment explaining how qs is used and whether is must be non-nil.